File: superstitioMain/java/superstitio/cardModifier/modifiers/card/InsideEjaculationTag.java
package superstitio.cardModifier.modifiers.card;

import com.megacrit.cardcrawl.cards.AbstractCard;
import superstitio.DataManager;
import superstitio.customStrings.stringsSet.UIStringsSet;

public class InsideEjaculationTag extends AbstractCardTagModifier {
    public static final String ID = DataManager.MakeTextID(InsideEjaculationTag.class);
    private static final UIStringsSet uiStrings = InsideEjaculationTag.getUIStringsWithSFW(ID);

    public static AbstractCard.CardTags getInsideEjaculationTag() {
        return DataManager.CardTagsType.InsideEjaculation;
    }

    @Override
    protected UIStringsSet getUiStrings() {
        return uiStrings;
    }

    @Override
    protected String getID() {
        return ID;
    }

    @Override
    protected AbstractCard.CardTags getTag() {
        return getInsideEjaculationTag();
    }
}

File: superstitioMain/java/superstitio/cardModifier/modifiers/card/InsideEjaculationTag.kt
package superstitio.cardModifier.modifiers.card

import com.megacrit.cardcrawl.cards.AbstractCard
import superstitio.DataManager
import superstitio.DataManager.CardTagsType
import superstitio.customStrings.stringsSet.UIStringsSet

class InsideEjaculationTag : AbstractCardTagModifier()
{
    override val uiStrings: UIStringsSet
        get() = Companion.uiStrings

    override fun getID(): String
    {
        return ID
    }

    override val tag: AbstractCard.CardTags
        get() = getInsideEjaculationTag()

    companion object
    {
        protected val ID: String = DataManager.MakeTextID(InsideEjaculationTag::class.java)
        private val uiStrings: UIStringsSet = getUIStringsWithSFW(ID)

        fun getInsideEjaculationTag(): AbstractCard.CardTags = CardTagsType.InsideEjaculation
    }
}
